tcp/client/conn.go

```diff
@@ -59,6 +60,7 @@ func NewConn(
 	connection *coapNet.Conn,
 	createBlockWise func(cc *Conn) *blockwise.BlockWise[*Conn],
 	inactivityMonitor InactivityMonitor,
+	requestMonitor RequestMonitorFunc,
```

Comment:
This breaks the signature of the public API, we should avoid that in non major library updates. (Same with `NewSession` in `tcp/client/session.go`  and `NewConn` in `udp/client/conn.go`).

The signatures of the NewConn/NewSession should remain the same as before, but we need new functions that take requestMonitor argument.

@jkralik do we prefer something like:

```go
func NewConnWithRequestMonitor(connection *coapNet.Conn, createBlockWise func(cc *Conn) *blockwise.BlockWise[*Conn], inactivityMonitor InactivityMonitor, requestMonitor RequestMonitorFunc, cfg *Config)
{
...
}
```

or perhaps we should go for function with variadic number of options?

```go
type Option struct {
	CreateBlockWise func(cc *Conn) *blockwise.BlockWise[*Conn]
	InactivityMonitor InactivityMonitor
        RequestMonitor RequestMonitorFunc
}

func NewConnWithOpts(connection *coapNet.Conn, cfg *Config, opts ...Option)
{
...
}
```

I would prefer the variadic version.